spec/requests/api/registrations_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe 'User Registration', type: :request do
+  let(:headers) { { HTTP_ACCEPT: 'application/json' } }
+
+  context 'with valid credentials' do
+    it 'returns a user and token' do
+      post '/api/auth', params: {email: 'me@mail.com',
+                                 password: 'whatever',
+                                 password_confirmation: 'whatever',
+                                 first_name: 'John',
+                                 last_name: 'Doe',
+                              }, headers: headers
+
```

Comment:
I would extract JSON.parse(response.body) to a helper method. A bit WET to repeat that in every test.

Reply:
Fixed, thanks for guidance